soccer/gameplay/evaluation/touchpass_positioning.py

```diff
@@ -0,0 +1,100 @@
+import robocup
+import constants
+import main
+import evaluation.passing
+
+## The Touchpass Positioner finds the best location within a rectangle
+#
+# Example usage:
+# tpos = TouchpassPositioner()
+# best_pt = tpos.evaluate(ball.pos, robot2.pos)
+class TouchpassPositioner:
+
+    def __init__(self):
+        self.debug = False
+
+    # Defaults to False
+    # if True, uses the system state drawing methods to draw Windows
+    @property
+    def debug(self):
+        return self._debug
+    @debug.setter
+    def debug(self, value):
+        self._debug = value
+
+    ## Returns a robocup.Rect object that is the default location to be evaluated
+    #
+    # Takes a current ball position/initial kick position (robocup.Point)
+    def generate_default_rectangle(self, kick_point):
+        offset_from_edge = 0.3
+
+        if kick_point.x > 0:
+            # Ball is on right side of field
+            toReturn = robocup.Rect(robocup.Point(0, min(constants.Field.Length - offset_from_edge, main.ball().pos.y - 0.3)),
+                    robocup.Point(-constants.Field.Width / 2 + offset_from_edge, min(constants.Field.Length * 3 / 4, main.ball().pos.y - 2)))
+        else:
+            # Ball is on left side of field
+            toReturn = robocup.Rect(robocup.Point(0, min(constants.Field.Length - offset_from_edge, main.ball().pos.y - 0.3)),
+                    robocup.Point(constants.Field.Width / 2 - offset_from_edge, min(constants.Field.Length * 3 / 4, main.ball().pos.y - 2)))
+        return toReturn
+
+    ## Returns a list of robocup.Point object that represent candidate points. Takes in a robocup.Rect
+    def get_points_from_rect(self, rect, threshold=0.50):
+        outlist = []
+        currentx = rect.min_x()
+        currenty = rect.max_y()
+
+        # Loop through from top left to bottom right
+
+        while currenty > rect.min_y():
+            while currentx < rect.max_x():
+                candiate = robocup.Point(currentx, currenty)
+                if not constants.Field.TheirGoalShape.contains_point(candiate):
+                    outlist.extend([candiate])
+                currentx = currentx + threshold
+            currenty = currenty - threshold
+            currentx = rect.min_x()
+        return outlist
+
+    ## Evaluates a single point, and returns the probability of it making it.
+    def eval_single_point(self, kick_point, receive_point):
```

Comment:
It'd be good to clarify that this is the probability of the pass AND the shot succeeding.  Also clarify that the probability is between zero and one.

Having values between zero and one for probabilities or power values is fairly standard throughout our codebase, but I think there may still be a few places where it's 0 to 100.  Those should be changed though :)